### PR TITLE
Added support for custom Navigator injection

### PR DESCRIPTION
**Pull request by peergynt**
GuiceNavigator can be extended and used for custom Navigator injection.

### CODE REVIEW

**Comment by berndhopp** [comment]
hmm, as it is now, an instance of GuiceNavigator would be assembled and used. How could you tell guice to assemble your custom class that derives from GuiceNavigator instead?

File: src/main/java/com/vaadin/guice/server/NavigatorManager.java [hunk 1]
```diff
@@ -61,14 +59,14 @@ void addNavigator(UI instance) {
                 defaultViewField.getName()
         );
 
-        Navigator navigator;
+        GuiceNavigator navigator = guiceVaadin.assemble(GuiceNavigator.class);
```
Comment (berndhopp) [inline src/main/java/com/vaadin/guice/server/NavigatorManager.java]: this is the line I was referring to in the comment. You would need to add another configuration option in order to tell guice which class is the actual Navigator to use. And then you can just remove the GuiceNavigator class altogether, you just need a parameter like Class<? extends Navigator> navigatorToUse that defaults to the standard com.vaadin.navigator.Navigator.

**Comment by peergynt** [comment]
Squashed all changes in one commit.
I made one more modification after your patch: in ReflectionUtils.java, it looks for the ViewContainer in the sub-classes until it gets to the UI class.

File: src/main/java/com/vaadin/guice/annotation/ViewContainer.java [hunk 2]
```diff
@@ -60,4 +63,8 @@
 @Retention(java.lang.annotation.RetentionPolicy.RUNTIME)
 @Documented
 public @interface ViewContainer {
+    /**
+     * the {@link Navigator} that is to be used for this ViewContainer
```
Comment (berndhopp) [inline src/main/java/com/vaadin/guice/annotation/ViewContainer.java:67]: my bad: should be {@link GuiceNavigator}

File: src/main/java/com/vaadin/guice/server/GuiceNavigator.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
+package com.vaadin.guice.server;
+
+import com.vaadin.navigator.Navigator;
+import com.vaadin.navigator.ViewDisplay;
+import com.vaadin.ui.ComponentContainer;
+import com.vaadin.ui.SingleComponentContainer;
+import com.vaadin.ui.UI;
+
+public class GuiceNavigator extends Navigator {
```
Comment (berndhopp) [inline src/main/java/com/vaadin/guice/server/GuiceNavigator.java:9]: a little javadoc would be nice here, like 'subclass this class if you want custom navigation behaviour'

**Pull request merged**